src/coupled/JAC/params/cable/grid_constants_cbl.F90

```diff
@@ -0,0 +1,57 @@
+!******************************************************************************
+! This source code is part of the Community Atmosphere Biosphere Land Exchange
+! (CABLE) model. This work is licensed under the CSIRO Open Source Software
+! License Agreement (variation of the BSD / MIT License).You may not use this
+! this file except in compliance with this License. A copy of the License is
+! available at https://trac.nci.org.au/trac/cable/wiki/license.
+!******************************************************************************
+
+MODULE grid_constants_mod_cbl
+
+!-----------------------------------------------------------------------------
+! Description:
+!   CABLE constants that define the "grid" ie. dimensions of arrays/vectors
+!   (replace constants USEd from elsewhere AS they are implemented)
+!   Potentially these dims could be read/set from CABLE nmls. However,
+!   NB. Simply changing these dims arbitrarily would render the model useless
+!
+! This MODULE is USEd throughout CABLE.
+!
+! Module specific documentation:https://trac.nci.org.au/trac/cable/wiki/TBC
+! Where it fits in the model flow:https://trac.nci.org.au/trac/cable/wiki/TBC
+!******************************************************************************
+
+IMPLICIT NONE
+
+PUBLIC
+
+! Although it seems these model "dimensions" may indeed be be a configuration
+! variable that can/should be set thru a namelist, CABLE has been developed
+! (even at an algorithm level) with the implicit assumption of the values
+! defined here and is unlikely to yield meaningful results (if it runs at all)
+! should these be arbitrarily changed at runtime
+
+! # of land-cover/soil types. Types of land-cover for veg+nonveg i.e.(13+4)
+!-----------------------------------------------------------------------------
+! Req'd to be defined at compile time to read in pars. strictly speaking these
+! only need to be greater than ntiles, nsoils (below). However, there is no
+! point in allocating useless space here
+INTEGER, PARAMETER :: ntype_max = 17 ! Max # tiles ! compile time constant
+INTEGER, PARAMETER :: nsoil_max = 9  ! Max # soils ! req'd to read in pars
+INTEGER, PARAMETER :: nsl       = 6  ! # soil layers !sm_levels in JULES IO
+INTEGER, PARAMETER :: nsnl      = 3  ! # snow layers
+INTEGER, PARAMETER :: nrb       = 3  ! # rad bands VISual/NIR + Legacy incl LW
+INTEGER, PARAMETER :: nrs       = 4  ! # streams (VIS+NIR)*(Direct+Diffuse)=4
+INTEGER, PARAMETER :: nsCs      = 2  ! # soil carbon stores
```

Comment:
why is this 2? there are 3 soil carbon stores.

Reply:
In CASA there are 3 soil carbon stores (fast, slow, passive) - but those variables are defined with `msoil=3`.

`nsCs` is used in the `bgc%` TYPE - and co-opted by `SLI`/`landuse3` - and it could be that it is supposed to be 2.  

Perhaps leave for now.

Reply:
I've checked in offline. landuse3 is using `ncs=2` from `cable_define_types.F90` only for the `bgc%` variable and it is using `msoil` from casadimension for the casa variables. So it's ok to keep the value 2 for now. Making it clearer, dropping deprecated feature would be welcome but not necessary here.

I couldn't find anything about this in SLI.

Reply:
> I couldn't find anything about this in SLI.

Looking again you're likely correct - what I thought I'd spotted wasn't a direct usage of `nsCs` but something like an assumed shape, via `ncs->nsCs=2`. 

There is usage of `var(i)%csoil(:,1)` and `var(i)%csoil(:,2)` and similar in SLI - but at that point `%csoil` is a heat capacity of soil, not a soil carbon variable. Importantly it has dimension `(1:mp,1:ms)` (so number of points, number of soil layers) with these set by `cable_define_types`.

However, I'll keep an eye out - for this.

ps. I agree it would be nice to remove `cable_carbon` and its dependencies